python/adelphi/adelphi/nb.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import yaml
+
+from itertools import chain
+
+from adelphi.exceptions import KeyspaceSelectionException, TableSelectionException
+from adelphi.export import BaseExporter
+
+SEQS={}
+SEQS["text"] = "Mod(1000000000); ToString() -> String"
+DISTS={}
+DISTS["text"] = "Hash(); Mod(1000000000); ToString() -> String"
+
+log = logging.getLogger('adelphi')
+
+# A collection of functionally static functions
+def dist_binding_name(col):
+    return "{}_dist".format(col.name)
+
+
+def seq_binding_name(col):
+    return "{}_seq".format(col.name)
+
+
+def build_bindings(table):
+    rv = {}
+    for (col_name, col) in table.columns.items():
+        if col.cql_type not in DISTS.keys():
+            raise ColumnTypeException("No nosqlbench distribution configured for type {}".format(col.cql_type))
+        rv[dist_binding_name(col)] = DISTS[col.cql_type]
+
+        # If we have a primary key we'll also need a seq for insert statements
+        if col_name in [c.name for c in table.primary_key]:
+            if col.cql_type not in SEQS.keys():
+                raise ColumnTypeException("No nosqlbench sequence configured for type {}".format(col.cql_type))
+            rv[seq_binding_name(col)] = SEQS[col.cql_type]
+
+    return rv
+
+
+def build_select_statements(keyspace, table):
+    key_bindings = " and ".join(["{} = {}".format(key.name, "{" + dist_binding_name(key) + "}") for key in table.primary_key])
```

Comment:
here we probably have to quote upper/mixed-case entity names (unless NoSQLBench handles it internally) otherwise C* will complain

Reply:
Good point; I wondered a bit about that when I was putting this code together and forgot to follow up with it.  I'll do some more testing to see if nosqlbench does the right thing... although I'll prolly just wind up quoting it here anyway.

Reply:
Latest commit added explicit quoting for keyspace, table and column names in generated statements.  A run with nb appears to be okay.